tests/kafkatest/tests/consumer_test.py

```diff
@@ -292,28 +231,75 @@ def test_broker_failure(self, clean_shutdown, enable_autocommit):
+    def test_group_consumption(self):
+        """
+        Verifies correct group rebalance behavior as consumers are started and stopped. 
+        In particular, this test verifies that the partition is readable after every
+        expected rebalance.
 
-        consumer = self._setup_consumer(self.STOPIC)
-        producer = self._setup_producer(self.STOPIC, max_messages=total_records)
+        Setup: single Kafka cluster with a group of consumers reading from one topic
+        with one partition while the verifiable producer writes to it.
 
-        partition = TopicPartition(self.STOPIC, 0)
+        - Start the consumers one by one, verifying consumption after each rebalance
+        - Shutdown the consumers one by one, verifying consumption after each rebalance
+        """
+        consumer = self.setup_consumer(self.TOPIC)
+        producer = self.setup_producer(self.TOPIC)
 
-        consumer.start()
-        self._await_all_members(consumer)
+        partition = TopicPartition(self.TOPIC, 0)
 
         producer.start()
-        wait_until(lambda: producer.num_acked == total_records, timeout_sec=20,
-                   err_msg="Producer failed waiting for messages to be written")
 
-        wait_until(lambda: consumer.last_commit(partition) == total_records, timeout_sec=10,
-                   err_msg="Consumer failed to read all expected messages")
+        num_started = 0
+        for node in consumer.nodes:
+            consumer.start_node(node)
+            num_started += 1
+            self.await_members(consumer, num_started)
+            self.await_consumed_messages(consumer)
 
-        assert consumer.current_position(partition) == total_records
+        num_stopped = 0
+        for node in consumer.nodes:
+            consumer.stop_node(node)
+            num_stopped += 1
 
-    def test_valid_assignment(self):
-        consumer = self._setup_consumer(self.TOPIC)
-        consumer.start()
-        self._await_all_members(consumer)
-        assert self._valid_assignment(consumer.current_assignment())
+            if num_stopped < self.num_consumers:
+                self.await_members(consumer, self.num_consumers - num_stopped)
+                self.await_consumed_messages(consumer)
+
+        assert consumer.current_position(partition) == consumer.total_consumed(), \
+            "Total consumed records did not match consumed position"
+
+        assert consumer.last_commit(partition) == consumer.current_position(partition), \
+            "Last committed offset did not match last consumed position"
+
+
+class AssignmentValidationTest(VerifiableConsumerTest):
+    TOPIC = "test_topic"
+    NUM_PARTITIONS = 6
+
+    def __init__(self, test_context):
+        super(AssignmentValidationTest, self).__init__(test_context, num_consumers=3, num_producers=0,
+                                                num_zk=1, num_brokers=2, topics={
+            self.TOPIC : { 'partitions': self.NUM_PARTITIONS, 'replication-factor': 2 },
```

Comment:
Does replication factor matter in this test at all? I can't think of why it would matter for the replication factor, so it seems weird to make it 2 here.